v2/tools/generator/internal/codegen/pipeline/create_arm_types_test.go

```diff
@@ -119,6 +123,149 @@ func TestCreateFlattenedARMTypeWithResourceRef_CreatesExpectedConversions(t *tes
 	g.Expect(err).ToNot(HaveOccurred())
 	state, err = simplify.Run(ctx, state)
 	g.Expect(err).ToNot(HaveOccurred())
+	state, err = strip.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	test.AssertPackagesGenerateExpectedCode(t, state.Definitions())
+}
+
+func TestCreateFlattenedARMTypeWithConfigMap_CreatesExpectedConversions(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	// Define a test resource
+	specProperties := test.CreateObjectDefinition(
+		test.Pkg2020,
+		"PersonProperties",
+		test.FullNameProperty,
+		test.FamilyNameProperty,
+		test.KnownAsProperty)
+	specPropertiesProp := astmodel.NewPropertyDefinition(
+		"Properties",
+		"properties",
+		specProperties.Name()).SetFlatten(true).MakeTypeOptional()
+	spec := test.CreateSpec(test.Pkg2020, "Person", specPropertiesProp, test.NameProperty)
+	status := test.CreateStatus(test.Pkg2020, "Person")
+	resource := test.CreateARMResource(test.Pkg2020, "Person", spec, status, test.Pkg2020APIVersion)
+
+	defs := make(astmodel.TypeDefinitionSet)
+	defs.AddAll(resource, status, spec, specProperties, test.Pkg2020APIVersion)
+
+	idFactory := astmodel.NewIdentifierFactory()
+	omc := config.NewObjectModelConfiguration()
+	g.Expect(
+		omc.ModifyProperty(
+			specProperties.Name(),
+			test.FullNameProperty.PropertyName(),
+			func(pc *config.PropertyConfiguration) error {
+				pc.SetInputConfigMapMode(config.InputConfigMapModeRequired)
+				return nil
+			})).
+		To(Succeed())
+	g.Expect(
+		omc.ModifyProperty(
+			specProperties.Name(),
+			test.FamilyNameProperty.PropertyName(),
+			func(pc *config.PropertyConfiguration) error {
+				pc.SetInputConfigMapMode(config.InputConfigMapModeOptional)
+				return nil
+			})).
+		To(Succeed())
+
+	configuration := config.NewConfiguration()
+	configuration.ObjectModelConfiguration = omc
+
+	addConfigMaps := AddConfigMaps(configuration)
+	createARMTypes := CreateARMTypes(idFactory)
+	applyARMConversionInterface := ApplyARMConversionInterface(idFactory)
+	flatten := FlattenProperties()
+	simplify := SimplifyDefinitions()
+	strip := StripUnreferencedTypeDefinitions()
+
+	// Don't need a context when testing
+	state := NewState().WithDefinitions(defs)
+	ctx := context.TODO()
+	state, err := addConfigMaps.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+	state, err = createARMTypes.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+	state, err = applyARMConversionInterface.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+	state, err = flatten.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+	state, err = simplify.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
+	state, err = strip.Run(ctx, state)
+	g.Expect(err).ToNot(HaveOccurred())
```

Comment:
Use `RunTestPipeline`

```suggestion
	state, err := RunTestPipeline(
		NewState().WithDefinitions(defs),
		addConfigMaps,
		createARMTypes,
		applyARMConversionInterface,
		flatten,
		simplify,
		strip)
```

Reply:
Nice!

Reply:
Fixed in all tests here.